src/socket/mod.rs

```diff
@@ -0,0 +1,58 @@
+use bcc_sys::bccapi::bpf_prog_type_BPF_PROG_TYPE_SOCKET_FILTER as BPF_PROG_TYPE_SOCKET_FILTER;
+use std::collections::HashMap;
+use std::iter::Iterator;
+
+use crate::core::BPF;
+use crate::error::BccError;
+
+#[derive(Debug, Default)]
+pub struct Socket {
+    handler: Option<String>,
+    ifaces: Vec<String>,
+}
+
+impl Socket {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn handler(mut self, handler: &str) -> Self {
+        self.handler = Some(handler.to_owned());
+        self
+    }
+
+    pub fn iface(mut self, iface: &str) -> Self {
```

Comment:
This function name doesn't convey that it is additive. `add_interface(..)` and `add_interfaces(..)` might be better names